ControllerDriver/networkhandler.cpp

```diff
@@ -83,26 +92,71 @@ bool isConnected() {
   return (wStatus == WL_CONNECTED);
 }
 
-template <typename T>
-void swapEndian(T &val) {
-    union U {
-        T val;
-        std::array<std::uint8_t, sizeof(T)> raw;
-    } src, dst;
-
-    src.val = val;
-    std::reverse_copy(src.raw.begin(), src.raw.end(), dst.raw.begin());
-    val = dst.val;
+template <>
+void swapEndian<uint32_t>(uint32_t &val) {
+    val = htonl(val);
+}
+template <>
+void swapEndian<uint16_t>(uint16_t &val) {
+    val = htons(val);
+}
+template <>
+void swapEndian<uint8_t>(uint8_t &val) {
+    // do nothing.
+}
+template <>
+void swapEndian<int8_t>(int8_t &val) {
+    // do nothing.
 }
 
 template <typename T>
-inline void swapEndianIfNeeded(T &val) {
-  // ifdef to determine byte order? ESP32 and Arduino is little-endian and needs a byte swap for network byte order.
+void writeNumberToBuffer(T val, uint8_t* buf, const uint16_t &pos) {
   swapEndian(val);
+  memcpy(&buf[pos], (uint8_t*) &val, sizeof(T));
 }
 
 template <typename T>
-void writeNumberToBuffer(T val, uint8_t* buf, const uint16_t &pos) {
-  swapEndianIfNeeded(val);
-  memcpy(&buf[pos], (uint8_t*) &val, sizeof(T));
+T readNumberFromBuffer(uint8_t* buf, const uint16_t &pos) {
+  T result;
+  memcpy((uint8_t*) &result, &buf[pos], sizeof(T));
+  swapEndian(result);
+  return result;
+}
+
+void checkMessages() {
+  bool newData = false;
+  while (client.available()) {
+    int bytes = client.read(readBuf, readBufferSize);
+    if (bytes <= 0) {
+      break;
+    } else {
+      newData = true;
+    }
+    
+    if (inputMaxLength < inputCurrentLength + bytes) { // resize buffer if needed
+      uint8_t* tmp = inputBuffer;
+      inputMaxLength = inputCurrentLength + bytes + inputSizeStep;
+      inputBuffer = new uint8_t[inputMaxLength];
+      if (tmp != NULL) {
+        memcpy(inputBuffer, tmp, inputCurrentLength);
+      }
```

Comment:
Since you allocated `new` memory for the buffer, you should `delete []` the old memory referenced by `tmp`.  Otherwise, you'll have a significant memory leak.  
```suggestion
        delete [] tmp;
      }
```